inference-engine/thirdparty/clDNN/src/layout_optimizer.cpp

```diff
@@ -842,6 +985,106 @@ impl_types layout_optimizer::get_preferred_impl_type(program_node& node) {
+            bool first_conv = input_layout.size.feature[0] <= 4;
+            bool enable_onednn_dw_fp16_conv = fp16_input && is_depthwise;
+            if (((has_groups && !enable_onednn_dw_fp16_conv) || first_conv) &&
+                (conv.get_output_layout().format == format::b_fs_yx_fsv16)) {
+                preferred_impl = impl_types::ocl;
+            }
+        }
+
+        if (node.is_type<deconvolution>()) {
+            auto& deconv = node.as<deconvolution>();
+            auto input_layout = deconv.input().get_output_layout();
+            bool valid_ic = input_layout.size.feature[0] >= 16;
+            bool valid_groups = deconv.get_primitive()->groups == 1;
+            bool valid_post_ops = /* get_post_ops_count(node) <= 10 */ false;
+            bool valid_batch = input_layout.size.batch[0] < 16;  // oneDNNs optimized kernel doesn't support big batches yet
+            bool valid_params = valid_ic && valid_groups && valid_post_ops && valid_batch;
+            if (!valid_params)
+                preferred_impl = impl_types::ocl;
+        }
+
+        // [WA] onednn doesn't support > 32 post-ops. Remove once onednn improve post-ops for GPU.
+        if (/* get_post_ops_count(node) > 32 */ true) {
+           preferred_impl = impl_types::ocl;
+        }
+
+        if (!data_types_are_suitable_for_onednn(node)) {
+            preferred_impl = impl_types::ocl;
+        }
+
+        for (auto& fo : node.get_fused_primitives()) {
+            if (fo.node->is_type<eltwise>()) {
+                auto in_layout = node.get_dependency(fo.dep_start_idx).get_output_layout();
+                auto out_layout = node.get_output_layout();
+                auto in_dt = in_layout.data_type;
+                auto out_dt = out_layout.data_type;
+                if ((out_layout.count() == in_layout.count()) &&
+                    (data_type_traits::is_floating_point(in_dt) || data_type_traits::is_floating_point(out_dt)) && in_dt != out_dt) {
+                    preferred_impl = impl_types::ocl;
+                    break;
+                }
+            } else if (fo.node->is_type<activation>()) {
+                // Some activations aren't implemented in oneDNN
+                auto activation_prim = fo.node->as<activation>().get_primitive();
+                if (activation_prim->activation_function == activation_func::negative ||
+                    activation_prim->activation_function == activation_func::negation ||
+                    activation_prim->activation_function == activation_func::sign)
+                    preferred_impl = impl_types::ocl;
+            }
+        }
+
+        // oneDNN doesn't support asymmetric weights quantization
+        if (node.is_type<convolution>() && node.as<convolution>().weights_zero_points_term())
+            preferred_impl = impl_types::ocl;
+
+        // OneDNN doesn't support sum post ops for deconvolutions
```

Comment:
Maybe it makes sence to use unified style in your comments, doesn't it?
Currently: line 1076 - OneDNN, line 1072 - oneDNN, lines 1042, 1016 - onednn